### DISCUSSION TITLE
Please DO NOT post images of code, data, error messages, etc.

### DISCUSSION THREAD

**Discussion by n4n0b1t3** [smartcontractkit/full-blockchain-solidity-course-py#1176] [General]
# How do I ask a good question?
_source: adapted from [StackOverflow - How do I ask a good question?](https://stackoverflow.com/help/how-to-ask)_

We’d love to help you. To improve your chances of getting an answer, here are some tips:

## Search, and research
First, please search this discussion group. There is a high chance, others had exactly the same problem as you do now. 

## Write a title that summarizes the specific problem
- **Always start with the lesson** you are currently working on. Even though your question might be of a more general nature, it helps answerers to understand what concepts you might not yet be familiar with.

**Examples**
Bad: I don't know what is wrong?
Good: Lesson 3: I cannot install pipx with yarn?

Bad: Need help with nonetype error
Good: Lesson 6: TypeError: argument of type 'NoneType' is not iterable

## Introduce the problem before you post any code
In the body of your question, start by expanding on the summary you put in the title. Explain how you encountered the problem you're trying to solve, and any difficulties that have prevented you from solving it yourself.

## Help others reproduce the problem
Not all questions benefit from including code, but if your problem is _with code_ you've written, you should include some. But don't just copy in your entire program! Not only is this likely to get you in trouble if you're posting your private key, it likely includes a lot of irrelevant details that readers will need to ignore when trying to reproduce the problem. Here are some guidelines:
- Include just enough code to allow others to reproduce the problem.
- Always format your code by using the <> symbol of the editor.  
- **DO NOT post images of code, data, error messages, etc.** - copy or type the text into the question. 

## Proofread before posting!
Now that you're ready to ask your question, take a deep breath and read through it from start to finish. Pretend you're seeing it for the first time: does it make sense? Try reproducing the problem yourself, in a fresh environment, and make sure you can do so using only the information included in your question. Add any details you missed. Now is a good time to make sure that your title still describes the problem!

## Mark answered questions
OK, you are lucky, someone answered your question and might have helped you to solve a problem. Maybe you solved it yourself, this is great! To allow others who might have the same issue to see, if your question was answered, you need to mark it as such.

**Comment by n4n0b1t3** [comment]
@cromewar Can you please pin this article to the top. Reading screenshots makes my eyes bleed.

**Reply by cromewar** [reply]
Is pinned now! and this is great I was writing an article just like this haha
This will help for the next PL on which and making a compilation of all guides and tips ;).

**Comment by PatrickAlphaC** [comment]
Great post here!! Thank you so much!!
